docs/elasticsearch-implementation.md

```diff
@@ -0,0 +1,71 @@
+# ElasticSearch
+
+We use ElasticSearch 7.4 and the [django-elasticsearch-dsl](https://django-elasticsearch-dsl.readthedocs.io/en/latest/) library to interact with ElasticSearch. 
+
+## Implementing ElasticSearch for a Wagtail Page
+
+In order to introduce consistency across the different apps within consumerfinance.gov we are encouraging the following structure for interacting with ElasticSearch. We are looking to organize all of our search related code to a sub directory within the respective apps models directory. A typical structure would look like `ask-cfpb/models/search/documents.py`, where ask-cfpb is the app where the search will reside in. Within the documents.py file we would expect to see the following classes.
+
+### Wagtail Page
+
+The first class to consider is the wagtail page, which is what our content editors will see and use from within the CMS. This is where we define all of our wagtail fields and should align with any other wagtail pages we leverage.
```

Comment:
I think we've decided against expecting the Wagtail page itself to live with the Elasticsearch Document, haven't we? We just need to ensure we don't have any circular dependencies between the `Document` class and other models that might use it.

Reply:
There was review comment in the Ask ES7 PR that requested `documents.py` be moved out of the model search directory. The `documents.py` file just contains the `AnswerPageDocument` class and is located as follows: `ask_cfpb/documents.py`. The Ask ES7 PR review also requested that the `AnswerPageSearch` class be moved into `cfgov/search` so that it could be made more generic and that way any of our searches could pass a `Document` subclass